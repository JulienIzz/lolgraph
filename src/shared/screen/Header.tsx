import styled from "@emotion/native";
import { theme } from "../theme/theme";

interface Props {
  title: string;
  backButton?: boolean;
}

export const Header = ({ title, backButton }: Props) => {
  return (
    <Background>
      <TitleContainer>
        {backButton ? <BackButton /> : null}
        <Title>{title}</Title>
      </TitleContainer>
    </Background>
  );
};

const BackButton = () => {
  return (
    <BackButtonTouchable>
      <BackButtonImage
        source={require("../../../assets/utils/arrow_left.png")}
      />
    </BackButtonTouchable>
  );
};

const Background = styled.SafeAreaView({
  backgroundColor: theme.primary,
  width: "100%",
});

const TitleContainer = styled.View({
  width: "100%",
  height: 60,
  justifyContent: "center",
  alignItems: "center",
});

const Title = styled.Text({
  fontWeight: "bold",
  fontSize: 20,
});

const BackButtonTouchable = styled.TouchableOpacity({
  height: 30,
  width: 30,
  position: "absolute",
  top: 15,
  left: 15,
});

const BackButtonImage = styled.Image({
  flex: 1,
  width: "100%",
  height: "100%",
});
