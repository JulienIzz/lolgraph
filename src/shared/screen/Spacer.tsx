import { View } from "react-native";

interface Props {
  vertical?: number;
  horizontal?: number;
}

export const Spacer = ({ vertical, horizontal }: Props) => {
  return <View style={{ height: vertical, width: horizontal }} />;
};
