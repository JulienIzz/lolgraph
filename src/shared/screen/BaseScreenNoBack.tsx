import styled from "@emotion/native";
import { Header } from "./Header";

interface ScreenProps {
  children: React.ReactNode;
  title: string;
}

export const BaseScreenNoBack = ({ children, title }: ScreenProps) => {
  return (
    <Container>
      <Header title={title} />
      {children}
    </Container>
  );
};

const Container = styled.View({
  flex: 1,
});
