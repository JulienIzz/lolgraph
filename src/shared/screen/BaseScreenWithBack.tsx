import styled from "@emotion/native";
import { Header } from "./Header";
import { QueryBoundaries } from "../utils/QueryBoundaries";

interface ScreenProps {
  children: React.ReactNode;
  title: string;
}

export const BaseScreenWithBack = ({ children, title }: ScreenProps) => {
  return (
    <Container>
      <Header backButton title={title} />
      <QueryBoundaries>{children}</QueryBoundaries>
    </Container>
  );
};

const Container = styled.View({
  flex: 1,
});
