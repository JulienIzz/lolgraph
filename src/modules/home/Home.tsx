import styled from "@emotion/native";
import { BaseScreenNoBack } from "../../shared/screen/BaseScreenNoBack";
import { HomeButton } from "./HomeButton";
import { useNavigation } from "@react-navigation/native";

export type HomeRouteProps = undefined;

export const Home = () => {
  const navigation = useNavigation();
  return (
    <BaseScreenNoBack title="MyLEAGUE">
      <Container>
        <HomeButton
          label="Profile"
          icon={require("../../../assets/utils/user.png")}
          onPress={() => navigation.navigate("ProfileInput")}
        />
        <HomeButton
          label="Timers"
          icon={require("../../../assets/utils/clock.png")}
          onPress={() => navigation.navigate("ProfileInput")}
        />
      </Container>
    </BaseScreenNoBack>
  );
};

const Container = styled.View({
  flex: 1,
  justifyContent: "center",
  alignItems: "center",
});
