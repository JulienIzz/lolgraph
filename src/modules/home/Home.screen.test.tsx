import { render, screen } from "@testing-library/react-native";
import { Home } from "./Home.screen";
import { NavigationContainer } from "@react-navigation/native";
import { SafeAreaProvider } from "react-native-safe-area-context";

jest.mock(
  "react-native-safe-area-context",
  // eslint-disable-next-line @typescript-eslint/no-var-requires
  () => require("react-native-safe-area-context/jest/mock").default,
);

jest.mock("@react-navigation/native", () => ({
  ...jest.requireActual("@react-navigation/native"),
  useNavigation: () => ({ navigate: jest.fn() }),
}));

test("home screen with title is correctly rendered", async () => {
  const expectedHeaderTextForTest = "MyLEAGUE";

  render(
    <SafeAreaProvider>
      <NavigationContainer>
        <Home />
      </NavigationContainer>
    </SafeAreaProvider>,
  );

  const HeaderOutput = screen.getByText(expectedHeaderTextForTest);

  expect(HeaderOutput).toBeTruthy();
  expect(screen.toJSON()).toMatchSnapshot();
});
