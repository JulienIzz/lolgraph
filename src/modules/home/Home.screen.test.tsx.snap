// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home screen with title is correctly rendered 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    insets={
      {
        "bottom": 0,
        "left": 0,
        "right": 0,
        "top": 0,
      }
    }
    style={
      [
        {
          "backgroundColor": "#00B4D8",
          "height": 60,
          "paddingTop": 0,
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 20,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        MyLEAGUE
      </Text>
    </View>
  </View>
  <View
    style={
      [
        {
          "alignItems": "center",
          "flex": 1,
          "margin": 25,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        {
          "height": 75,
          "width": undefined,
        }
      }
    />
    <TextInput
      onChangeText={[Function]}
      placeholder="Summoner's name"
      placeholderTextColor="gray"
      style={
        [
          {
            "borderColor": "gray",
            "borderRadius": 6,
            "borderWidth": 1,
            "fontSize": 16,
            "height": 40,
            "textAlign": "center",
            "width": 250,
          },
          undefined,
        ]
      }
    />
    <View
      style={
        {
          "height": 12,
          "width": undefined,
        }
      }
    />
    <View
      onLayout={[Function]}
      style={
        {
          "borderColor": "gray",
          "borderRadius": 6,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "width": 250,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 35,
            "justifyContent": "space-between",
          }
        }
      >
        <Text
          style={
            [
              {
                "flex": 1,
                "fontSize": 16,
                "writingDirection": "ltr",
              },
              {
                "color": "gray",
                "textAlign": "center",
              },
              {},
            ]
          }
        >
          Server
        </Text>
        <Image
          source={1}
          style={
            {
              "height": 20,
              "tintColor": "gray",
              "width": 20,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        {
          "height": 40,
          "width": undefined,
        }
      }
    />
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#00B4D8",
          "borderColor": "gray",
          "borderRadius": 6,
          "borderWidth": 1,
          "height": 40,
          "justifyContent": "center",
          "opacity": 1,
          "width": 250,
        }
      }
    >
      <Text
        style={
          [
            {
              "fontSize": 16,
              "fontWeight": "bold",
            },
            undefined,
          ]
        }
      >
        Search
      </Text>
    </View>
  </View>
</View>
`;
